analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShadowsOuterStaticMember.cs

```diff
@@ -151,39 +138,36 @@ private static void CheckEvent(SymbolAnalysisContext context, INamedTypeSymbol c
                     return eventSyntax?.Identifier.GetLocation();
                 },
                 "event");
-        }
 
-        private static void CheckField(SymbolAnalysisContext context, INamedTypeSymbol containerClassSymbol, IFieldSymbol field)
-        {
-            CheckFieldOrProperty(field, containerClassSymbol, context,
-                f =>
-                {
-                    var reference = f.DeclaringSyntaxReferences.FirstOrDefault();
-                    var syntax = reference?.GetSyntax() as VariableDeclaratorSyntax;
-                    return syntax?.Identifier.GetLocation();
-                },
+        private static void CheckField(SymbolAnalysisContext context, INamedTypeSymbol containerClassSymbol, IFieldSymbol field) =>
+            CheckFieldOrProperty(field, containerClassSymbol, context, f =>
+                 {
+                     var reference = f.DeclaringSyntaxReferences.FirstOrDefault();
+                     var syntax = reference?.GetSyntax() as VariableDeclaratorSyntax;
+                     return syntax?.Identifier.GetLocation();
+                 },
                 "field");
-        }
 
-        private static void CheckProperty(SymbolAnalysisContext context, INamedTypeSymbol containerClassSymbol, IPropertySymbol property)
-        {
+        private static void CheckProperty(SymbolAnalysisContext context, INamedTypeSymbol containerClassSymbol, IPropertySymbol property) =>
             CheckFieldOrProperty(property, containerClassSymbol, context, p =>
             {
                 var reference = p.DeclaringSyntaxReferences.FirstOrDefault();
                 if (reference == null)
                 {
                     return null;
                 }
-                if (!(reference.GetSyntax() is PropertyDeclarationSyntax syntax))
+                if (reference.GetSyntax() is not PropertyDeclarationSyntax syntax)
```

Comment:
The `Func<ISymbol, Location> locationSelector` parameter is defined quite a lot here for all kinds of symbols. We could implement it as an extension method so it could be reused in other places too:
```CS
public static SyntaxToken? GetIdentifierToken(this ISymbol symbol) =>
  symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() switch
  {
    PropertyDeclarationSyntax p => p.Identifier,
    VariableDeclaratorSyntax v => v.Identifier,
    // ..
   // Could be extended to other declarations too
   MethodDeclarationSyntax m => m.Identifier,
    _ => null,
  };
```
By doing so, the `Func<ISymbol, Location> locationSelector` parameter is not needed anymore. It could also be reused in `CheckNamedType` to get rid of the switch statement there.